lib/sshv2.c

```diff
@@ -988,7 +988,9 @@ sshv2_buffer_get_int64 (gftp_request * request, sshv2_message * message,
   if ((ret = sshv2_buffer_get_int32 (request, message, 0, 0, &lonum)) < 0)
     return (ret);
 
-  snum = (gint64) lonum | ((gint64) hinum >> 32);
+  snum = (guint64) lonum | ((guint64) hinum << 32);
```

Comment:
How about casting both of these to off_t instead of guint64? Good catch with the bit shift error.